File: day-86/_308_Bubble_Sort.cpp
// _308_Bubble_Sort.cpp

// https://www.geeksforgeeks.org/problems/bubble-sort/1

class Solution
{
public:
    // Function to sort the array using bubble sort algorithm.
    void bubbleSort(vector<int> &arr)
    {
        int n = arr.size();
        for (int i = 0; i < n; i++)
        {
            for (int j = 1; j < n; j++)
            {
                if (arr[j] < arr[j - 1])
                {
                    swap(arr[j], arr[j - 1]);
                }
            }
        }
    }
};

// Input: arr[] = [4, 1, 3, 9, 7]
// Output: [1, 3, 4, 7, 9]

File: day-86/_308_Bubble_Sort.py
# _308_Bubble_Sort.py

# https://www.geeksforgeeks.org/problems/bubble-sort/1

class Solution:
    #Function to sort the array using bubble sort algorithm.
    def bubbleSort(self,arr):
        # code here
        for i in range(0,len(arr)):
            for j in range(1,len(arr)):
                if arr[j]<arr[j-1]:
                    arr[j-1],arr[j] = arr[j],arr[j-1]

# Input: arr[] = [4, 1, 3, 9, 7]
# Output: [1, 3, 4, 7, 9]